src/Data/Monoid/Extra.hs

```diff
@@ -15,3 +19,16 @@ import Data.Monoid
 -- > mwhen False "test" == ""
 mwhen :: Monoid a => Bool -> a -> a
 mwhen b x = if b then x else mempty
+
+-- | The intersection of 'Data.Monoid.Endo' and 'Control.Arrow.Kleisli'. This
+--   type provides a 'Monoid' instance for composition of monadic actions
+--   @a -> m a@.
+newtype KEndo m a = KEndo { appKEndo :: a -> m a }
```

Comment:
This can be generalized to 
```
newtype Endomorphism k a = Endomorphism (k a a)

instance Category k => Semigroup (Endomorphism k a) where 
  (<>) = coerce (Control.Category..)

instance Category k => Monoid (Endomorphism k a) where 
  mempty = coerce Control.Category.id
```

Then `Endomorphism k a` would cover both usual `Endo` (when `k ~ (->)`) and monadic endo (when `k ~ Kleisli`).

Reply:
This seems like a great idea. That said, once we have gone this far, I wonder if extra is the right place for it? It seems to be more categorical and less compose a few obvious things?